app/src/main/java/com/example/plantsapp/data/firebase/repository/FirebaseUserRepository.kt

```diff
@@ -0,0 +1,21 @@
+package com.example.plantsapp.data.firebase.repository
+
+import com.example.plantsapp.domain.model.User
+import com.example.plantsapp.domain.repository.UserRepository
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class FirebaseUserRepository @Inject constructor() : UserRepository {
+
+    override var user: User? = null
+    private set
+
+    override fun setUser(user: User) {
+        this.user = user
+    }
+
+    override fun requireUser(): User = user!!
```

Comment:
also - instead of using !! that throws just NullPointerException, you can throw exceptions that will be more clear about situation that occurs: 

For excemple - IllegalStateException would work great here - because you try to ask requireUser when state is Incorrect